modules/tftp/server.rb

```diff
@@ -108,6 +108,30 @@ def pxeconfig_file(mac)
     end
   end
 
+  class Pxegrub2targetos < Server
+    def setup_bootloader(mac, os)
+      pxeconfig_dir_mac = pxeconfig_dir(mac)
+      FileUtils.mkdir_p(config_dir)
+      bootloader_directory = (Proxy::TFTP::Plugin.settings.bootloader_universe || "/usr/local/share/bootloader-universe")
+      FileUtils.cp("#{bootloader_directory}/#{os}/grubx64.efi", "#{pxeconfig_dir_mac}/grubx64.efi")
```

Comment:
1. the file copy will only work if the directory exists
2. what happens if the directory and/or its files did already exist?

Reply:
1. The directory is created by `FileUtils.mkdir_p(pxeconfig_dir_mac)`.
2. The files are overridden. This is ok.